docs/topics/custom-images.md

```diff
@@ -0,0 +1,130 @@
+# Custom images
+
+This document will help you get a CAPZ Kubernetes cluster up and running with your custom image.
+
+## Reference images
+
+An *image* defines the operating system and Kubernetes components that will populate the disk of each node in your cluster.
+
+By default, images offered by "capi" in the [Azure Marketplace][azure-marketplace] are used. You can list these *reference images* with this command:
+
+```bash
+az vm image list --publisher cncf-upstream --offer capi --all -o table
+```
+
+But for more control over your nodes, you should build, publish, and use a *custom image*.
+
+## Building a custom image
+
+Cluster API uses the Kubernetes [Image Builder][image-builder] tools. You should use the [Azure images][image-builder-azure] from that project as a starting point for your custom image.
+
+[The Image Builder Book][capi-images] explains how to build the images defined in that repository, with instructions for [Azure CAPI Images][azure-capi-images] in particular.
+
+### Operating system requirements
+
+For your custom image to work with cluster API, it must meet the operating system requirements of the bootstrap provider. For example, the default `kubeadm` bootstrap provider has a set of [`preflight checks`][kubeadm-preflight-checks] that a VM is expected to pass before it can join the cluster.
+
+### Kubernetes version requirements
+
+The reference images are each built to support a specific version of Kubernetes. When using your custom images based on them, take care to match the image to the `version:` field of the `KubeadmControlPlane` and `MachineDeployment` in the YAML template for your workload cluster.
+
+To upgrade to a new Kubernetes release with custom images requires this preparation:
+
+- create a new custom image which supports the Kubernetes release version
+- update the `image:` section for `AzureMachineTemplate`s in the YAML template to reference the new custom image
```

Comment:
instead of updating the image directly, the instructions in https://cluster-api.sigs.k8s.io/tasks/kubeadm-control-plane.html#upgrading-the-control-plane-machines recommend first creating a new template and then modifying the template ref and the Version in one apply to avoid triggering two updates.

`In order to only trigger a single upgrade, the new MachineTemplate should be created first and then both the Version and InfrastructureTemplate should be modified in a single transaction.`